lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -42,23 +84,41 @@ typedef _KeyDataResponseCallback = void Function(bool handled);
 class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   /// Private constructor, since only dart:ui is supposed to create one of
   /// these.
-  EnginePlatformDispatcher._() {
+  EnginePlatformDispatcher() {
     _addBrightnessMediaQueryListener();
+    _addHighContrastMediaQueryListener();
     _addFontSizeObserver();
+    registerHotRestartListener(dispose);
   }
 
   /// The [EnginePlatformDispatcher] singleton.
   static EnginePlatformDispatcher get instance => _instance;
-  static final EnginePlatformDispatcher _instance =
-      EnginePlatformDispatcher._();
+  static final EnginePlatformDispatcher _instance = EnginePlatformDispatcher();
 
   /// The current platform configuration.
   @override
   ui.PlatformConfiguration configuration = ui.PlatformConfiguration(
     locales: parseBrowserLanguages(),
     textScaleFactor: findBrowserTextScaleFactor(),
+    accessibilityFeatures: setHighContrastonAccessibilityFeatures(),
   );
 
+  /// Initialize accessibility features based on the current value of high contrast flag
+  static EngineAccessibilityFeatures setHighContrastonAccessibilityFeatures() {
```

Comment:
I suggest giving this method a generic name like `initializeAccessibilityFeatures`. We may use it to initialize other fields as well.